### DISCUSSION TITLE
noun_chunks function return a empty list

### DISCUSSION THREAD

**Discussion by zyxdSTU** [explosion/spaCy#6337] [Language Support]
I use a  chinese model  called zh-core-web-sm2.3.1，
and when I call the nonu_chunks function to get nouns phase, I  just find get a empty list.
so someone know the  right way to get nouns phases.

**Comment by adrianeboyd** [comment]
The `noun_chunks` property currently returns an empty list if the noun chunks method isn't implemented. It would probably be better to raise a `NotImplementedError` because this is confusing. We can think about changing this for the upcoming v3.

If you'd like noun chunks, you'd need to implement the noun chunks iterator. Here's the English implementation as an example:

https://github.com/explosion/spaCy/blob/45c9a688285081cd69faa0627d9bcaf1f5e799a1/spacy/lang/en/syntax_iterators.py

We'd be happy to accept a PR that added this for Chinese!

**Comment by gleidsonh** [comment]
@adrianeboyd I'm trying to implement this on Brazilian Portuguese (I believe lang/pt), but it's not working.

I copy exact english syntax_iterators.py, compile spacy again (I don't know if it's really necessary) but not works. Chunks is still empty. Am I doing right way?

**Comment by adrianeboyd** [comment]
You also have to import it in `__init__.py` to set the default syntax iterators, like this:

https://github.com/explosion/spaCy/blob/cd61d264ef4dbae642713169ade41404c24335df/spacy/lang/en/__init__.py#L29

(If you're just changing `.py` files and you have a editable install, you shouldn't need to recompile it.)

Because English uses a different set of dependency labels (not UD), you'd be better off starting with another language that uses UD labels as an example, maybe French is a good example? (I think the French one is simpler than the Spanish one, which uses a different approach that I'm not sure is 100% working as intended at this point.)

**Comment by gleidsonh** [comment]
@adrianeboyd Thank you so much! I just need to improve, but at least it's working.

In fact, Spanish is more similar to Portuguese than French.

Thanks again!

**Comment by adrianeboyd** [comment]
Yes, Spanish is more similar, but I think that the French noun chunks iterator may be a better starting point. I don't think the noun chunk definition is going to be that different here, since it's mostly about the UD dependencies. You can test both, though. In particular with the Spanish version, check carefully to make sure it's not producing overlapping noun chunks when you have nested NPs.
